### DISCUSSION TITLE
Automerge: preset vs. packageRules

### DISCUSSION THREAD

**Discussion by mbrodala** [renovatebot/renovate#11354] [Request Help] [answered]
There seem to be 2 ways to set up automerge e.g. for patch updates:

1. [Set `automerge` option via a package rule](https://docs.renovatebot.com/automerge-configuration/#automerge-non-major-updates).
2. Use the [`:automergePatch` preset](https://docs.renovatebot.com/presets-default/#automergepatch)

The latter expands to:

```json
{
  "separateMinorPatch": true,
  "patch": {
    "automerge": true
  },
  "pin": {
    "automerge": true
  },
  "lockFileMaintenance": {
    "automerge": true
  }
}
```

Why are there `pin` and `lockFileMaintenance` in this preset instead of only `patch`?

We don't use `lockFileMaintenance` but `"rangeStrategy": "update-lockfile"`, will this lead to issues or undesired updates? (The same goes for `pin` BTW.)

**Comment by rarkins** [comment, marked as answer]
Pinning and lock file maintenance are considered less "risky" than a patch update, so they were included in that preset. I think that preset may have existed from before we even used `packageRules` for this type of thing.

**Reply by mbrodala** [reply]
Thanks for the clarification. So to be 100% sure you'd suggest to _not_ use the preset but directly use a package rule instead?

**Reply by rarkins** [reply]
Assuming you don't want to automerge pin and lockFileMaintenance, use a package rule instead. I'd be happy to change this one (and presumably automergeMinor too) in the next major release.
